### DISCUSSION TITLE
Sonoff S26R2ZB terrible performance

### DISCUSSION THREAD

**Discussion by pvprodk** [Koenkk/zigbee2mqtt#10909] [Devices]
Anyone else having terrible performance with Sonoff S26R2ZB ?

I've purchased two of these, which are advertised to be routers/range extenders, but they dont seem to allow end devices to join, even if I hold the end device right next to the S26R2ZB, the end device still joins directly to the coordinator. If i try to add end devices which are out of range of the coordinator but in range of the S26R2ZB it fails to join.

On top of this, the range is very bad, and very low LQI reported.

**Comment by Koenkk** [comment]
> Anyone else having terrible performance with Sonoff S26R2ZB ?

yes

- https://github.com/Koenkk/zigbee2mqtt/issues/10559
- https://github.com/Koenkk/zigbee2mqtt/issues/10282

**Reply by pvprodk** [reply]
Thank you very much.
Tried searching before posting, but guess I only searched the discussions. My bad.

**Reply by peteralm097** [reply]
Hi

Itead support have confirmed to me that firmware 2.0.0 for these plugs is defective and they are offering me replacements with a newer firmware if I pay them shipping... (not sure I have to pay shipping to replace a defective item tho...) I have suggested that they could contact Koenkk with the firmware for an OTA. I have an open ticket number if Koenkk wants to get in touch with me.

**Reply by euantennant** [reply]
I also have an open ticket with Itead Support for devices with firmware 2.0.0 and 2.0.1. Itead claim that the issue is addressed with firmware 2.1.0.

**Reply by Hedda** [reply]
Hop enough customers ask for their support so that ITead releases an OTA firmware image compatible with Zigbee standard format!

This is discussed further in https://github.com/Koenkk/zigbee2mqtt/issues/10282

**Comment by Hedda** [comment]
Hmm, wondering if the issue is radio module hardware antenna design in it or just firmware that ITead is shipping with Sonoff S26R2ZB?

https://itead.cc/product/sonoff-s26r2zb-zigbee-smart-plug-uk-de-fr/

https://sonoff.tech/product/smart-plug/s26r2zb/

I understand that the S26R2ZB contains a new CC2652 based radio module with PCB circuit board antenna, (probably designed by [CoolKit](https://www.coolkit.cn)).

@[beezly](https://github.com/beezly) posted pictures https://github.com/Koenkk/zigbee2mqtt/issues/10282 which seem to show S26R2ZB contain a not previously seen CC2652P (CC2652 P1F) based radio module that looks to be using a ceramic chip antenna instead of a PCB circuit board antenna, that radio board also look to have no label markings?

![image](https://user-images.githubusercontent.com/6320001/152678463-edd0ebcb-7346-407f-baaf-70727f977744.png)

![image](https://user-images.githubusercontent.com/6320001/152678505-8e551736-e07b-4f3d-954a-e2feadd1f64d.png)

![image](https://user-images.githubusercontent.com/6320001/152678518-3f26959d-15f6-4ee3-826f-89a1ca49cf6d.png)

![image](https://user-images.githubusercontent.com/6320001/152678529-3cd0a05c-9868-4957-b896-13e144a59800.png)

I am also wondering if that is a similar Zigbee radio module design by CoolKit that will be used in the upcoming ITead Sonoff ZBBridge-P?

https://github.com/Koenkk/zigbee2mqtt/discussions/10758

FCC pictures show ITead Sonoff ZBBridge-P using Texas Instruments CC2652P (CC2652P1) MCU chip radio module saying “SM-031 v1.1”:

![image](https://user-images.githubusercontent.com/6320001/149344517-dd8226e1-5476-46a7-a1d0-b6dd3ac78318.png)

![image](https://user-images.githubusercontent.com/6320001/149344589-c148f43f-299f-4b81-b117-a90932889e8f.png)

CoolKit Technologies is the company that makes their EFR32MG21 based “SM-011 V1.0” module for their previous ITead Sonoff ZBBridge:

https://github.com/CoolKit-Technologies/DevDocs/tree/master/Zigbee

https://www.coolkit.cn/product/sm-011/

And @CoolKit-Technologies also designed ITead's first Silicon Labs EFR342MG21 based Zigbee 3.0 USB Dongle which hardware design was seriously criticized for having a poorly designed/tuned PCB circuit board antenna with not enough components to make reception less susceptible to electromagnetic interference:

https://itead.cc/product/zigbee-3-0-usb-dongle/

**Reply by Hedda** [reply]
FYI, in related news ITead also announced ITead also announced Sonoff ZBMINI-L No-Neutral Zigbee 3.0 Smart Switch which contains a presumably CoolKit designed “**SM-029 V1.1**” radio module based on Texas Instruments based CC2652 (which has been confirmed to use the same CC2652P chip as their [“Sonoff Zigbee 3.0 USB Dongle Plus” adapter](https://github.com/Koenkk/zigbee2mqtt/discussions/8840)) and it has been reported to ship with its firmware configured as an "End-Device" and not as as "Zigbee Router":

https://community.home-assistant.io/t/sonoff-zbmini-l-zigbee-3-0-smart-switch-by-itead-works-without-neutral-wire-and-based-on-cc2652p-so-should-be-a-good-zigbee-router/384644/

![image](https://user-images.githubusercontent.com/6320001/152678132-2f0447a0-93a5-42f5-882c-c33eb6ffcf5e.png)

![image](https://user-images.githubusercontent.com/6320001/152678253-054e2f63-e81e-464a-9fed-75e1962a4db0.png)

![image](https://user-images.githubusercontent.com/6320001/152678266-05863a3a-1bb0-45d8-89d5-a08873bc68de.png)

![image](https://user-images.githubusercontent.com/6320001/152678262-e33815ec-9806-4e2b-85d6-1d44e41e9d6c.png)

![image](https://user-images.githubusercontent.com/6320001/152678288-27292ecd-ffc9-406a-a3ee-a2d9195f44ef.png)

**Comment by Hedda** [comment]
FYI, Posted tip yesterday in https://github.com/Koenkk/zigbee2mqtt/issues/10282 that might be a good idea to try PTVO firmware but **select the CC2652R chip type (and not CC2652P) in the PTVO firmware configurator** as the  S26R2ZB radio board module has probably been optimised for use without a power amplifier enabled as per https://ptvo.info/custom-firmware-for-sonoff-zbmini-01minizb-and-ewelink-switch-zr-02-923/

That article is the conclusion from -> https://github.com/ptvoinfo/zigbee-configurable-firmware/issues/96

Manually configuring the transmit power to 5 dBm or less in the firmware should probably also work?

PS: To manually flash new firmware to CC2652 based devices like these you need cJTAG (c/JTAG) compatible adapter like TI CC-DEVPACK-DEBUG adapter or Texas Instruments XDS110. See -> https://electrolama.com/radio-docs/advanced/flash-jtag/

**Reply by Koenkk** [reply]
@ptvoinfo a bit offtopic, but I just noticed https://ptvo.info/cc2652r1-cc2652rb-cc2652p-cc1352p2-configurable-router-firmware-748/ . Does your firmware support binding? If yes, then this would be a great switch to put behind a wall controlling zigbee bulbs via a group.

**Reply by Hedda** [reply]
@Koenkk suggest post that question via new issue to -> https://github.com/ptvoinfo/zigbee-configurable-firmware/issues/

**Reply by ptvoinfo** [reply]
@Koenkk Yes, all versions of my firmware supports incoming and outgoing binding for OnOff and LevelControl clusters.

**Comment by renatoyamane** [comment]
Hi,
Is there any news about an workaround to make these (not so)smart plugs Sonoff S26R2ZBTPG working?

My gateway is the Sonoff NSPanel PRO;
My sensors are Sonoff SNZB-01/02/03/04;
They are disconnecting frequently, so I added three routers S26R2ZBTPG and them more 4!

I'm living in a flat (60m2) with a lot of sensors and "_smart_" plugs that are disconnecting frequently.

**Comment by Piipz** [comment]
How do you find out the firmware of S26R2ZB plugs?
I have been messing around with them for a year and always thought it's the problem with the house (concrete walls etc.) but this here tells me its the plugs. I'm thinking of replacing them but i would need to know what version I have and probably would need a newer version then? Help appreciated :)

**Reply by ChrisLane** [reply]
You can find the firmware version by clicking into the device on your Zigbee2Mqtt dashboard.
When I click on mine for example I see the following among other details:
```
Firmware build date    20210511
Firmware version    2.0.0
```

**Comment by sharkzor** [comment]
I just got this plug with firmware 2.1.1
Also has frequent disconnects.
Unusable!

**Reply by redpixy** [reply]
yes I also have the same problem with these plugs. Not sure if other experience the following. The plug seems to disrupt my entire mesh. it causes other devices to be disconnected and my entire networks to be unresponsive.

**Comment by renatoyamane** [comment]
I forgot to report here:
Sonoff replaced my S26R2ZBTPG with BASICZBR3 many months ago.
Waste of time... these devices also don't work as signal extenders.
